### ISSUE TITLE
Android&iOS using of this library

### ISSUE THREAD

**Issue by ProgrammingLife** [maps4print/azul#253]
Can Azul be used in Android & iOS development?

**Comment by fschutt** [comment]
Technically yes, but it's untested for that purpose. The windowing backend supports mobile, and WebRender supports mobile, too.

I'm not sure whether it works, because I haven't tried getting it to run on mobile. You might find [these instructions](https://github.com/rust-windowing/winit#android) helpful.

**Comment by ProgrammingLife** [comment]
@fschutt sounds great! And what about iOS? The same way?

**Comment by fschutt** [comment]
Yes, the same. It may work, but I can't test it. 

Windows - Mac - Linux Desktop and WASM are the targets right now.

**Comment by ProgrammingLife** [comment]
@fschutt guys, look... you definetely should try to make it run under Android and iOS. Pretty sure there are a lot of opportunities and scope of application on those markets specially for Azul. Not many really fast and lightweight App engines there on mobile. Flutter, Native, React Native (lol) and that's all. If Azul would create builds for mobiles that would be really stunning.

**Comment by dumblob** [comment]
I wonder if there are any CI providers of free Android & iOS plans this project could leverage to automatically build it upon each commit on these platforms.

Does anyone know about such services?

This wouldn't substitute testing on these platforms but would IMHO significantly lower the bar for adoption by newcomers for basically zero effort from @fschutt and co.

**Comment by fschutt** [comment]
@dumblob CicleCI provides testing for Android and iOS for winit AFAIK. Also, since Azul is 99% Rust (only a bit of C), it should be easy to cross-compile.

However, right now I'm only working on getting it 100% running on Windows first, every other platform can come later. I simply don't have the time right now to support 5+ platforms and four compilation targets (C, C++, Python, Rust) alone. While the API itself is good (the architecture is ok, I've tested Azul in practice), it may still take a while unti a 1.0 (Windows-Mac-Linux-only) release.

**Comment by dumblob** [comment]
Makes sense to first target the "primary" platforms and after some minimal "MVP-like" stabilization jump on Android & iOS CI (and if there is commercial interest, build a community around full-featured Android & iOS support instead of just plain CI building).
